hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+                              'The reconstruction will be performed with the first values')
+                sb_position_temp = sb_position.inav[0].data
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_position_temp = sb_position.data
+        else:
+            sb_position_temp = sb_position.deepcopy()
+        #
+
+        # Parsing sideband size
+        if sb_size is None:  # Default value is 1/2 distance between sideband and central band
+            if reference is None:
+                sb_size = self.find_sideband_size(sb_position)
+            else:
+                sb_size = reference.find_sideband_size(sb_position)
+        else:
+            if not isinstance(sb_size, BaseSignal):
+                if isinstance(sb_size, np.ndarray) and sb_size.size > 1:  # transpose if np.array of multiple instances
+                    sb_size = BaseSignal(sb_size).T
+                else:
+                    sb_size = BaseSignal(sb_size)
+
+        if sb_size.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_size.axes_manager.navigation_size:
+                warnings.warn('Sideband size dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_size_temp = np.float64(sb_size.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_size_temp = np.float64(sb_size.data)
+        else:
+            sb_size_temp = sb_size.deepcopy()
+        #
+
+        # Standard edge smoothness of sideband aperture 5% of sb_size
+        if sb_smoothness is None:
+            sb_smoothness = sb_size * 0.05
+        else:
+            if not isinstance(sb_smoothness, BaseSignal):
+                if isinstance(sb_smoothness, np.ndarray) and sb_smoothness.size > 1:
+                    sb_smoothness = BaseSignal(sb_smoothness).T
+                else:
+                    sb_smoothness = BaseSignal(sb_smoothness)
+
+        if sb_smoothness.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_smoothness.axes_manager.navigation_size:
+                warnings.warn('Sideband smoothness dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_smoothness_temp = np.float64(sb_smoothness.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_smoothness_temp = np.float64(sb_smoothness.data)
+        else:
+            sb_smoothness_temp = sb_smoothness.deepcopy()
+
+        # Convert sideband size from 1/nm or mrad to pixels
+        if sb_unit == 'nm':
+            sb_size_temp = sb_size_temp / np.mean(self.f_sampling)
```

Comment:
Actually the reconstruction is designed for square images only at the moment. I thought that it will deliver reasonable results for ( m, n | x, y) images where x=~y but it seem it doesn't really work. @FWin22 could you please check if `f_sampling` and other parts are capable for x!=y? Though I tend to restrict current version to square images only and do non-square once in next PR...